### DISCUSSION TITLE
[2023-06-19] GUI Team Demo Day

### DISCUSSION THREAD

**Discussion by farmaazon** [enso-org/enso#7051] [Weekly Demo Day]
This week:
* I was carefully reviewing Wojciech's PR with special attention to the soundness of the PR (no such issues were found, however)
* Otherwise, I continued tracking the leaks after calling `drop` on our IDE wasm. It turned out that the Scene was leaking in [multiple places](https://github.com/enso-org/enso/pull/6911#issuecomment-1591059209), also found that our event listeners were not always detached on handler drop (because we haven't passed the options to removeEventListener method).
* In the process, I also make the error about visualization failure [disappear after some time](https://github.com/enso-org/enso/issues/6976)

**Comment by wdanilo** [comment]
I'm working on [new FRP integration with GUI](https://github.com/enso-org/enso/pull/7056). The current progress:

| Task  | Progress | Days Needed To Finish It |
| ------------- | ------------- | ------------- |
| Moving old FRP nodes to new impl  | 66% (66/99)  | 1 day |
| Testing all FRP nodes  | 0% (0/99)  | 1-2 days |
| Adding bridge network to the new impl | 0% | < 1 day |
| Moving `network::extend!` macro to new impl  | 0%  | 1 day |
| Heavy testing and debugging  | 0%  | 1-2 days |

To sum it up, I need another 4-6 days to properly finish and test the new implementation. There is a delay of ~5 days regarding the previous estimation that is caused by:
- Merging the [previous PR](https://github.com/enso-org/enso/pull/6944) taking more time (2-3 days more) than estimated. This was mainly caused by the amount of comments I got from @farmaazon (which I'm very thankful for, as they were very on-point).
- Me feeling a little bit unwell during Friday and Saturday.
- Technical problems with the integration – I was trying to do some of the trait implementations in a generic way, but it seems Rust's type system is not ready for it yet, so I created some magical macros to handle these cases instead.

**Comment by MichaelMauderer** [comment]
* Fixed "Fast typing leading to wrong component browser selection. #7032 

https://github.com/enso-org/enso/assets/1428930/dce82bb3-1c25-4a67-988a-d74469dc8fcd

* Fixed " Scrollbar not visible on Text visualization in Full Screen Mode" #6855 

https://github.com/enso-org/enso/assets/1428930/db80c5b7-69fd-4bf5-84ab-c83664227059

**Comment by vitvakatu** [comment]
- Minor refactorings in https://github.com/enso-org/enso/pull/6978 – CSS rules will no longer conflict with the rest of the application, on QA.
- Fix for cursor position when clicking into nodes (https://github.com/enso-org/enso/pull/7014) finally got merged.
- Fixed redundant imports when using dropdown widgets: https://github.com/enso-org/enso/pull/7028 (on QA)
- Bumping rustc to 1.70 (https://github.com/enso-org/enso/issues/6895) – in progress, moderate amount of new issues in the code.

Next week:
- https://github.com/enso-org/enso/issues/7047

**Comment by kazcw** [comment]
- Fixed dropdown laziness (#6865).
- Fixed a edge/expression edit logic bug (Case 1 in #7018).
- Fix regression in node selection rendering (2nd bug in #6975).

**Comment by mwu-tow** [comment]
* Integrating new remote logging endpoint; waited for a cloud-side fix that landed today, if everything works PR should be ready within a day.
* Redirecting logs to file: encountered some issues; should be ready within a day or two.

This week: finish ongoing work, fix undo/redo issues with visualizations.

**Comment by JaroslavTulach** [comment]
Engine team & performance:
- #7005 
- #7019

**Comment by Frizi** [comment]
- Still finishing PR with edge connection handling https://github.com/enso-org/enso/pull/7041. Still having issues with graph editor tests, specifically with mouse handling in them. Since I'm now using new mouse events API, the old hacky way of manipulating cursor position and sending events was not working properly. The edge bugs are fixed though.
- Going to continue working on design changes as soon as those tests are fixed (I hope still today).

https://github.com/enso-org/enso/assets/919491/49e560cb-0a29-4c6a-97ec-4370185b8c89

https://github.com/enso-org/enso/assets/919491/e62450ff-46b2-466f-ac33-f4f19e66ee1d
